src/value-parser.js

```diff
@@ -335,7 +335,8 @@ function valueParse(parser, metaData, options, callback) {
                     valueMetaData.collation = collation;
                     if (baseType === 0xA5 || baseType === 0xAD || baseType === 0xA7 || baseType === 0xAF || baseType === 0xE7 || baseType === 0xEF) {
                       return readDataLength(parser, valueMetaData.type, {}, null, (maxDataLength) => {
-                        valueMetaData.dataLength = maxDataLength;
+                        // skip the 2-byte max length sent for BIGVARCHRTYPE, BIGCHARTYPE, NVARCHARTYPE, NCHARTYPE, BIGVARBINTYPE and BIGBINARYTYPE types
+                        // and parse based on the length of actual data
```

Comment:
Sure, I'll push the tests after writing them in nodeunit. 

Some clarification - this PR is not using the globally defined MAX value. It uses [valueMetaData.dataLength](https://github.com/tediousjs/tedious/pull/596/files#diff-6ee9b395d11ea3cffaecde0296efd5aaR328), this is the data length derived from column header of the row.

Details on why this is preferred over the maxDataLength is in https://github.com/tediousjs/tedious/pull/596#discussion_r124680598. SqlClient too reads the maxDataLength(just to skip over it, never uses it), actual data is read using the length in column header.

This PR essentially would be something like this, if it was in structured programming

```node
if (baseType in BIGVARCHRTYPE || BIGCHARTYPE || NVARCHARTYPE || NCHARTYPE || BIGVARBINTYPE || BIGBINARYTYPE) {
  // ignore maxDataLength - we already have all necessary detail to call valueParse
  maxDataLength  = readDataLength(parser, valueMetaData.type, {}, null) 
}
valueParse(parser, valueMetaData, options, callback);
```